### PR TITLE
[Example] Use parsing correctly in candlestick example + remove unnecessary timeUnit, scale and axis

### PR DESCRIPTION
**Pull request by kanitw**

### CODE REVIEW

File: examples/specs/layer_candlestick.vl.json [hunk 1]
```diff
@@ -1,8 +1,13 @@
 {
   "$schema": "https://vega.github.io/schema/vega-lite/v2.json",
   "width": 320,
-  "description":"A candlestick chart inspired from Protovis(http://mbostock.github.io/protovis/ex/candlestick.html)",
+  "description": "A candlestick chart inspired from Protovis(http://mbostock.github.io/protovis/ex/candlestick.html)",
```
Comment (domoritz) [inline examples/specs/layer_candlestick.vl.json]: inpired by

File: examples/specs/layer_candlestick.vl.json [hunk 2]
```diff
@@ -1,8 +1,13 @@
...
   "data": {
+    "format": {
+      "parse": {
+        "date": "date:'%Y-%m-%d'"
```
Comment (domoritz) [inline examples/specs/layer_candlestick.vl.json]: Browsers should parse this automatically. Why do you need to add the format?

File: examples/specs/layer_candlestick.vl.json (hunk 1 above)
Comment (kanitw) [reply examples/specs/layer_candlestick.vl.json]: ?

File: examples/specs/layer_candlestick.vl.json (hunk 2 above)
Comment (kanitw) [reply examples/specs/layer_candlestick.vl.json]: Its parsing as utc

Comment (kanitw) [reply examples/specs/layer_candlestick.vl.json]: Vega & Vega-Lite simply uses JS parsing.  The behavior is consistent with the screenshot you send.  But time scale is default local time.  So if you want to visualize local time, need to parse correctly.

Comment (domoritz) [reply examples/specs/layer_candlestick.vl.json]: It's a bit weird that we `utc:FORMAT`, `utc`, and `date` parse as UTC but `date:FORMAT` parses as local, though.

File: examples/specs/layer_candlestick.vl.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "$schema": "https://vega.github.io/schema/vega-lite/v2.json",
   "width": 320,
-  "description": "A candlestick chart inspired from Protovis(http://mbostock.github.io/protovis/ex/candlestick.html)",
+  "description": "A candlestick chart inspired by Protovis (http://mbostock.github.io/protovis/ex/candlestick.html)",
```
Comment (domoritz) [inline examples/specs/layer_candlestick.vl.json]: maybe better say "inspired by an example in Protovis"

Comment (domoritz) [reply examples/specs/layer_candlestick.vl.json]: :-D

**Pull request merged**